parser/transcript/src/main/java/de/danoeh/antennapod/parser/transcript/SrtTranscriptParser.java

```diff
@@ -0,0 +1,126 @@
+
+public class SrtTranscriptParser {
+    private static String TAG = "SrtTranscriptParser";
+    // merge short segments together to form a span of 1 second
+    public static long minSpan = 1000L;
+
+    public static Transcript parse(String str) {
+        Transcript transcript = new Transcript();
+
+        List<String> lines = Arrays.asList(str.split("\n"));
+        Iterator<String> iter = lines.iterator();
+        String speaker = "";
+        StringBuffer body = new StringBuffer("");
+        String line;
+        String segmentBody = "";
+        long startTimecode = -1L;
+        long spanStartTimecode = -1L;
+        long endTimecode = -1L;
+        long duration = 0L;
+
+        while (iter.hasNext()) {
+            line = iter.next();
+
+            if (line.isEmpty()) {
+                continue;
+            }
+
+            if (line.contains("-->")) {
+                String[] timecodes = line.split("-->");
+                if (timecodes.length < 2) {
+                    continue;
+                }
+                startTimecode = parseTimecode(timecodes[0].trim());
+                endTimecode = parseTimecode(timecodes[1].trim());
+                if (startTimecode == -1 || endTimecode == -1) {
+                    continue;
+                }
+
+                if (spanStartTimecode == -1) {
+                    spanStartTimecode = startTimecode;
+                }
+                duration += endTimecode - startTimecode;
+                do {
+                    line = iter.next();
+                    if (StringUtil.isBlank(line)) {
+                        break;
+                    }
+                    body.append(line.strip());
+                    body.append(" ");
+                } while (iter.hasNext());
+            }
+
+            if (body.indexOf(":") != -1) {
+                String [] parts = body.toString().trim().split(":");
+                if (parts.length < 2) {
+                    continue;
+                }
+                speaker = parts[0];
+                body = new StringBuffer(parts[1].strip());
+            }
+            if (!StringUtil.isBlank(body.toString())) {
+                segmentBody += " " + body;
+                segmentBody = StringUtils.trim(segmentBody);
+                if (duration >= minSpan && endTimecode > spanStartTimecode) {
+                    Log.d(TAG, "SRT : " + Long.toString(spanStartTimecode) + " " + segmentBody);
+                    transcript.addSegment(new TranscriptSegment(spanStartTimecode,
+                            endTimecode,
+                            segmentBody,
+                            speaker));
+                    duration = 0L;
+                    spanStartTimecode = -1L;
+                    segmentBody = "";
+                }
+                body = new StringBuffer("");
+            }
+        }
+
+        if (! StringUtil.isBlank(segmentBody) && endTimecode > spanStartTimecode) {
```

Comment:
In other places of the code, we do not use a space between `!` and the thing to negate